### PR TITLE
Update AWS Lambda User Guide for Ballerina 1.2.8

### PR DESCRIPTION
**Pull request by lafernando**
$subject.

### CODE REVIEW

File: learn/deployment/aws-lambda.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ redirect_from:
 
 # AWS Lambda
 
-The AWS Lambda extension provides the functionality to expose a Ballerina function as an AWS Lambda function. This is done by importing the `ballerinax/awslambda` module and simply annotating the Ballerina function with the `awslambda:Function` annotation. Also, the Ballerina function must have the following signature: `function (awslambda:Context, json) returns json|error`. 
+The AWS Lambda extension provides the functionality to expose a Ballerina function as an AWS Lambda function. This is done by importing the `ballerinax/awslambda` module and simply annotating the Ballerina function with the `awslambda:Function` annotation. Also, the Ballerina function must have the following signature: `function (awslambda:Context, json|EventType) returns json|error`. The second parameter in the function is either a `json` or a domain specific event type, such as `awslambda:S3Event`. If the user does not provide a domain specific event type, the json representation of the incoming event will be sent to the function.
```
Comment (praneesha) [inline learn/deployment/aws-lambda.md:14]:
```suggestion
The AWS Lambda extension provides the functionality to expose a Ballerina function as an AWS Lambda function. This is done by importing the `ballerinax/awslambda` module and simply annotating the Ballerina function with the `awslambda:Function` annotation. Also, the Ballerina function must have the following signature: `function (awslambda:Context, json|EventType) returns json|error`. The second parameter in the function is either a `json` or a domain specific event type such as `awslambda:S3Event`. If the user does not provide a domain specific event type, the JSON representation of the incoming event will be sent to the function. 
```

File: learn/deployment/aws-lambda.md [hunk 2]
```diff
@@ -11,9 +11,9 @@ redirect_from:
...
+The AWS Lambda extension provides the functionality to expose a Ballerina function as an AWS Lambda function. This is done by importing the `ballerinax/awslambda` module and simply annotating the Ballerina function with the `awslambda:Function` annotation. Also, the Ballerina function must have the following signature: `function (awslambda:Context, json|EventType) returns json|error`. The second parameter in the function is either a `json` or a domain specific event type, such as `awslambda:S3Event`. If the user does not provide a domain specific event type, the json representation of the incoming event will be sent to the function. 
 
-The following Ballerina code gives an example on how to expose a function in AWS Lambda, which generates a SHA256 hash from the given input. 
+The following code presents a few examples on how to expose functions in AWS Lambda, which contains a generic `json` event input, and other functions which provides the event information using domain specific event types.
```
Comment (praneesha) [inline learn/deployment/aws-lambda.md:16]:
```suggestion
The following code presents a few examples on how to expose functions in AWS Lambda, which contain a generic `json` event input and other functions, which provide the event information using domain-specific event types.
```

File: learn/deployment/aws-lambda.md [hunk 3]
```diff
@@ -23,9 +23,34 @@ import ballerina/crypto;
...
+The first parameter with the [awslambda:Context](/learn/api-docs/ballerina/awslambda/objects/Context.html) object contains the information and operations related to the current function execution in AWS Lambda such as the request ID and the remaining execution time. The second parameter contains the input request data. This input value will vary depending on the source, which invoked the function e.g., an AWS S3 bucket update event. The return type of the function is `json|error`, which means in a successful scenario, the function can return a `json` value with the response or else in an error situation, the function will return an `error` value, which provides information on the error to the system. We can also provide functions which does not return anything at all, which implicitely signals a successful execution without a returning result.
```
Comment (praneesha) [inline learn/deployment/aws-lambda.md:53]:
```suggestion
The first parameter with the [awslambda:Context](/learn/api-docs/ballerina/awslambda/objects/Context.html) object contains the information and operations related to the current function execution in AWS Lambda such as the request ID and the remaining execution time. The second parameter contains the input request data. This input value will vary depending on the source, which invoked the function (e.g., an AWS S3 bucket update event). The return type of the function is `json|error`, which means in a successful scenario, the function can return a `json` value with the response or else in an error situation, the function will return an `error` value, which provides information on the error to the system. You can also provide functions, which do not return anything at all, which implicitly signals a successful execution without a returning result.
```

**Review by praneesha — APPROVED** [review]

**Comment by lafernando** [comment]
@praneesha This change should go to the website when the v1.2.8 is released, which will happen before the end of the week.

**Comment by praneesha** [comment]
> @praneesha This change should go to the website when the v1.2.8 is released, which will happen before the end of the week.

Ack. It will get auto synced to prod for the 1.2.8 release once we merge this PR.

**Comment by anupama-pathirage** [comment]
@praneesha Shouldn't we have this PR for dev repo first? IINM, we are not encouraged to send direct PRs to prod repo.

**Comment by praneesha** [comment]
> @praneesha Shouldn't we have this PR for dev repo first? IINM, we are not encouraged to send direct PRs to prod repo.

Yes. @lafernando - We need to send the PR to the "ballerina-dev-website" repo.

**Comment by praneesha** [comment]
The valid PR of this is https://github.com/ballerina-platform/ballerina-dev-website/pull/1215.

**Pull request closed without merging**